autoload/go/mod.vim

```diff
@@ -0,0 +1,128 @@
+    let l:tmpname = tr(l:tmpname, '\', '/')
+  endif
+
+  let current_col = col('.')
+  let l:args = ['go', 'mod', 'edit', '--fmt', l:tmpname]
+  let [l:out, l:err] = go#util#Exec(l:args)
+  let diff_offset = len(readfile(l:tmpname)) - line('$')
+
+  if l:err == 0
+    call go#mod#update_file(l:tmpname, fname)
+  else
+    let errors = s:parse_errors(fname, l:out)
+    call s:show_errors(errors)
+  endif
+
+  " We didn't use the temp file, so clean up
+  call delete(l:tmpname)
+
+  " Restore our cursor/windows positions.
+  call winrestview(l:curw)
+
+  " be smart and jump to the line the new statement was added/removed
+  call cursor(line('.') + diff_offset, current_col)
+
+  " Syntax highlighting breaks less often.
+  syntax sync fromstart
+endfunction
+
+" update_file updates the target file with the given formatted source
+function! go#mod#update_file(source, target)
+  " remove undo point caused via BufWritePre
+  try | silent undojoin | catch | endtry
+
+  let old_fileformat = &fileformat
+  if exists("*getfperm")
+    " save file permissions
+    let original_fperm = getfperm(a:target)
+  endif
+
+  call rename(a:source, a:target)
+
+  " restore file permissions
+  if exists("*setfperm") && original_fperm != ''
+    call setfperm(a:target , original_fperm)
+  endif
+
+  " reload buffer to reflect latest changes
+  silent edit!
+
+  let &fileformat = old_fileformat
+  let &syntax = &syntax
+
+  let l:listtype = go#list#Type("GoModFmt")
+
+  " the title information was introduced with 7.4-2200
+  " https://github.com/vim/vim/commit/d823fa910cca43fec3c31c030ee908a14c272640
+  if has('patch-7.4.2200')
+    " clean up previous list
+    if l:listtype == "quickfix"
+      let l:list_title = getqflist({'title': 1})
+    else
+      let l:list_title = getloclist(0, {'title': 1})
+    endif
+  else
+    " can't check the title, so assume that the list was for go fmt.
+    let l:list_title = {'title': 'Format'}
+  endif
+
+  if has_key(l:list_title, "title") && l:list_title['title'] == "Format"
+    call go#list#Clean(l:listtype)
+  endif
+endfunction
+
+" parse_errors parses the given errors and returns a list of parsed errors
+function! s:parse_errors(filename, content) abort
+  let splitted = split(a:content, '\n')
+
+  " list of errors to be put into location list
+  let errors = []
+  for line in splitted
+    let tokens = matchlist(line, '^\(.\{-}\):\(\d\+\):\(\d\+\)\s*\(.*\)')
+    if !empty(tokens)
+      call add(errors,{
+            \"filename": a:filename,
+            \"lnum":     tokens[2],
+            \"col":      tokens[3],
+            \"text":     tokens[4],
+            \ })
+    endif
+  endfor
+
+  return errors
+endfunction
+
+" show_errors opens a location list and shows the given errors. If the given
+" errors is empty, it closes the the location list
+function! s:show_errors(errors) abort
+  let l:listtype = go#list#Type("GoModFmt")
+  if !empty(a:errors)
+    call go#list#Populate(l:listtype, a:errors, 'Format')
```

Comment:
Instead of passing in the parsed errors and calling `go#list#Populate` manually, can you use an `errorformat` string and call `go#list#ParseFormat`?

Reply:
I just tried it, but I think it'll not work because we run the command on a temporary file first (because `go mod edit` does an in-place edit). This means that the initial `%f` is the temporary filename. This is copied from `fmt.vim` and it's the same issue. I'll check if I can do something ( I would prefer `ParseFormat` indeed)

Reply:
Hmmmm... If you know the temporary filename, then it should be easy enough to do a `substitute(...)` on `a:errors` before passing to `ParseFormat`.

Reply:
I think at this point, this is too much a hack. We're having a working version and it really works fine. Not sure if it's worth doing it (same applies for gofmt).